sentinel/src/main/kotlin/com/infinum/sentinel/ui/certificates/observer/SentinelWorkManager.kt

```diff
@@ -37,35 +39,38 @@ internal class SentinelWorkManager(
     }
 
     init {
-        WorkManager.initialize(
-            context,
-            Configuration.Builder()
-                .setMinimumLoggingLevel(android.util.Log.INFO)
-                .setWorkerFactory(workerFactory)
-                .build()
-        )
+        if (WorkManager.isInitialized().not()) {
+            WorkManager.initialize(
+                context,
+                Configuration.Builder()
+                    .setMinimumLoggingLevel(android.util.Log.INFO)
+                    .setWorkerFactory(workerFactory)
+                    .build()
+            )
+        }
     }
 
     @RequiresApi(Build.VERSION_CODES.O)
-    fun startCertificatesCheck(entity: CertificateMonitorEntity) =
+    fun startCertificatesCheck(entity: CertificateMonitorEntity) {
+
+        val delegatedWorkData = workDataOf(
+            WORKER_CLASS_NAME to CertificateCheckWorker::class.qualifiedName,
+            WORKER_ID to CertificateCheckWorker.NAME,
+            NOTIFY_INVALID_NOW to entity.notifyInvalidNow,
+            NOTIFY_TO_EXPIRE to entity.notifyToExpire,
+            EXPIRE_IN_AMOUNT to entity.expireInAmount,
+            EXPIRE_IN_UNIT to entity.expireInUnit.name
+        )
         WorkManager.getInstance(context)
             .enqueueUniquePeriodicWork(
-                CertificateCheckWorker.NAME,
+                DelegateWorker.DELEGATE_WORKER_ID,
                 ExistingPeriodicWorkPolicy.CANCEL_AND_REENQUEUE,
-                PeriodicWorkRequestBuilder<CertificateCheckWorker>(
+                PeriodicWorkRequestBuilder<DelegateWorker>(
```

Comment:
I can't connect pieces, why is this different than before? I mean in both cases, we pass the worker we want to run, shouldn't then even the previous line work fine?